students/ThanhNhan/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+import mailroom4 as mr4
+import os
+
+"""
+Send a thank you (adds a new donor or updates existing donor info)
+Create a report
+Send letters (creates files)
+Your unit tests should test the data manipulation logic code: generating thank you text, adding or updating donors, and listing donors.
+Send a thank you (adds a new donor or updates existing donor info)
+"""
+def test_existing_donor():
+
+    new_donor = {"Tom Cruise": [233.23, 12.17],
+             "Alyssa Nhan": [23., 23],
+             "Jessy Test": [33.0],
+             "Jin Unknown": [12.0, 12.0]
+             }
+    mr4.donor_dict = new_donor
+    new_donor_db = mr4.chk_exist_donor()
+    assert new_donor_db == ["Tom Cruise", "Alyssa Nhan", "Jessy Test", "Jin Unknown"]
+
+
+# Validate Name Input
+def test_validate_name():
```

Comment:
very nice!

though it's better to make these separate tests -- this way, if the first one fails, the rest won't run.

If this seems tedious -- it is. pytest provides a way to make a bunch of very similar tests liek this, called "parameterized tests" -- you'll learn about that next quarter.